polylogue/cli/click_app.py

```
@@ -0,0 +1,619 @@
+    console.print(group.get_help(root_ctx))
+    _print_command_listing(console, getattr(env.ui, "plain", False), entries)
+    _print_quick_examples(console)
+
+
+@click.group(context_settings={"help_option_names": ["-h", "--help"]})
+@click.option("--plain", is_flag=True, help="Force plain/CI-safe output even when running in a TTY")
+@click.option("--interactive", is_flag=True, help="Force interactive UI even when stdout/stderr are not TTYs")
+@click.pass_context
+def cli(ctx: click.Context, plain: bool, interactive: bool) -> None:
+    """Polylogue CLI (Click)."""
+    use_plain = plain or (not sys.stdout.isatty() or not sys.stderr.isatty()) and not interactive
```

Ruff (0.15.9)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)